Automattic/wp-calypso
# logmeinUrl package and usage

#### Changes proposed in this Pull Request

* Add `logmeingUrl` method to append the logmein parameter defined in D61461-code

#### Testing instructions

* ~Apply D61461-code~ (dependency, since deployed)
* Login to http://calypso.localhost:3000/, create a private site with a mapped domain (or log into my test user for `testing23465324891.home.blog/`)
* oauth flag set for testing without 3pc, and logmein flag set for testing this feature, e.g. visit `http://calypso.localhost:3000/?flags=oauth,logmein`
* Click here: ![image](https://user-images.githubusercontent.com/5436027/118505955-84e51280-b74a-11eb-8a6c-da4797a165ee.png)

You should end up the site in a logged in state with cookies against the mapped domain.

- Test both while logged out of the test site directly.
- Test sub.subdomain.blog domains include the param
- Test mapped.com domains include the param
- Test wpcomstaging urls do not include the param
- Test domain redirects do not include the param
- Test Atomic sites do not include the param

For testing in calypso using chrome with third party cookies disabled you'll need to use calypso's oauth mode, e.g. `http://calypso.localhost:3000/domains/manage/testing23465324891.home.blog?flags=oauth,logmein`.

Unit tests:
```
yarn test-client client/lib/logmein
```

Related to #52741 / #52740 / #52763
